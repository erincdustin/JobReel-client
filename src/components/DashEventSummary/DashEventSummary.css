.DashEventSummary > h4 {
  text-align: center;
}

.options > ul > li {
  margin: 10px;
  list-style: none;
  width: 90%;
}