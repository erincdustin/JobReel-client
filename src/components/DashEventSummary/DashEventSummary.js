import React from 'react';
import { Link } from 'react-router-dom';
import './DashEventSummary.css';
import jobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';

class DashEventSummary extends React.Component {

  static contextType = jobReelContext

  componentDidMount() {
    if (this.context.savedEvents.length === 0) {
      jobReelApiService.getSavedEvents()
        .then(res => {
          this.context.setSavedEvents(res);
        })
    }
  }

  renderEventSummaries = () => {
    let events = Array.from(this.context.savedEvents);
    events = events.sort((a, b) => {
      return new Date(a.date) - new Date(b.date);
    });
    events = events.slice(0, 3).map(event => {
      const date = new Date(event.date.slice(0, -1)).toDateString()
      return (
        <li key={event.event_id}>{event.event_name}, {date} - <a href={event.url}>{event.url}</a></li>
        )
      })
      return (
        <ul>
          {events}
        </ul>
    )
  }
  
  render() {
    return (
      <>
        <div className="DashEventSummary">
          <h4>{this.props.section}</h4>
        </div>
        <div className='options'>
            {this.renderEventSummaries()}
        </div>
        <Link to="/saved-events">View all</Link>
      </>
    )
  }
}

export default DashEventSummary;