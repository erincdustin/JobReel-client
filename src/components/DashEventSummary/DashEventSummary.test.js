import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PropTypes from 'prop-types'
import { BrowserRouter } from 'react-router-dom'
import DashEventSummary from './DashEventSummary'

class ContextProvider extends React.Component{
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      events: [1, 2, 3]
    }
  })

  render(){
    return this.props.children
  }
}

describe('DashEventSummary smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BrowserRouter><DashEventSummary /></BrowserRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})

describe('DashEventSummary snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><DashEventSummary /></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})