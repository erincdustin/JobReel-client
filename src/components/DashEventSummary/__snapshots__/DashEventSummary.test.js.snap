// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashEventSummary snapshot test renders the UI as expected 1`] = `<DashEventSummary />`;
